eclair-core/src/main/scala/fr/acinq/eclair/db/pg/PgChannelsDb.scala

```diff
@@ -67,16 +67,16 @@ class PgChannelsDb(implicit ds: DataSource, lock: PgLock) extends ChannelsDb wit
   override def addOrUpdateChannel(state: HasCommitments): Unit = withMetrics("channels/add-or-update-channel", DbBackends.Postgres) {
     withLock { pg =>
       val data = stateDataCodec.encode(state).require.toByteArray
-      using(pg.prepareStatement("UPDATE local_channels SET data=? WHERE channel_id=?")) { update =>
-        update.setBytes(1, data)
-        update.setString(2, state.channelId.toHex)
-        if (update.executeUpdate() == 0) {
-          using(pg.prepareStatement("INSERT INTO local_channels (channel_id, data, is_closed) VALUES (?, ?, FALSE)")) { statement =>
-            statement.setString(1, state.channelId.toHex)
-            statement.setBytes(2, data)
-            statement.executeUpdate()
-          }
-        }
+      using(pg.prepareStatement(
+        """
+          | INSERT INTO local_channels (channel_id, data, is_closed)
+          | VALUES (?, ?, FALSE)
+          | ON CONFLICT ON CONSTRAINT local_channels_pkey
```

Comment:
Is it guaranteed that the primary key will be named `local_channels_pkey`, regardless of the postgres version?

Reply:
I'm _almost certain_ (famous last words) that it is the case. However we can remove all doubt and explicitly name the constraint:
```sql
CREATE TABLE IF NOT EXISTS local_channels (channel_id TEXT NOT NULL CONSTRAINT local_channels_pkey PRIMARY KEY, data BYTEA NOT NULL, ...  
```

Reply:
NB: there is another variant that uses the column name, and I initially wanted to use that one, but couldn't figure out how to.

Reply:
Done with 8e7409e3f.